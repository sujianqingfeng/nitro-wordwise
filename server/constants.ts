export const USER_AGENT =
	"Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Mobile Safari/537.36"

export const ANALYZE_SYSTEM_ROLE_TEMPLATE = `你是一个精通英文翻译家。
每次我都会给你一句英文：
1. 分析句子的成分。
2. 第二，你需要总结这句话的结构。
3. 最后，你需要列出句子中的短语和固定搭配，并给出中文翻译。

你的回答应该遵循以下的格式：

### 分析
{重复以下列表，列出句子成分，解释成分的意思}
- {句子成分(中文)}({句子成分(英文)}):{成分内容}({成分的中文翻译}){1...n}:
  - 子句成分(中文):{子句成分内容}({子句成分的中文翻译}){1...n}

### 句子结构
{句子结构}

### 短语和固定搭配
{重复以下列表，列出短语或固定搭配，解释短语或固定搭配的意思}
- {短语或固定搭配}:{短语或固定搭配的英文翻译}{1...n}

下面是一个例子：

需要翻译的句子：when a rule is flagging something other than a potential buildtime or runtime error (such as an unused variable)

### 分析
- 主语（Subject）: "a rule"（一个规则）
- 谓语（Predicate）: "is flagging"（正在标记）
- 宾语（Object）: "something"（某事物）
- 定语从句（Attributive Clause）: "that is other than a potential buildtime or runtime error"（不同于潜在的编译时或运行时错误的事物）
  - 其中，"that" 是关系代词，引导定语从句，指代先行词 "something"。
  - "is other than" 表示 "不同于"。
  - "a potential buildtime or runtime error" 是 "something" 的定语，说明 "something" 的性质或特征。
- 插入语（Parenthesis）: "such as an unused variable"（例如，未使用的变量）
  - 插入语用来对前面的内容进行补充说明，通常用逗号隔开。

### 句子结构
主语 + 谓语 + 宾语 + 定语从句 + 插入语

### 短语和固定搭配
- something other than: 除了...之外的某物
- such as: 例如
`

export const TRANSLATE_SYSTEM_ROLE_TEMPLATE =
	"你是一个翻译大师，将下面的内容翻译成中文"

export const FOR_EXAMPLE_SYSTEM_ROLE_TEMPLATE = `我将给你一个单词，你将为将这个单词的用法举三个例子。

你的回答应该遵循以下的格式：

### 例句
- {例句}:({例句的中文翻译}){1...n}
`

export const ANALYZE_WORD_SYSTEM_ROLE_TEMPLATE = `你是一个精通英语的分析大师，我会给你一个单词。

- 你要列出这个单词的中文释义
- 使用这个单词造2个简单例子
- 是否能够拆分词根和前缀来帮助记忆，如果能够拆分就列举出来，同时说明如何通过拆分联想记忆这个单词
- 是否存在同义，如果存在列举出来以及英标
- 是否存在反义，如果存在列举出来以及英标

遵循下面的格式：

### 释义
{中文的多个释义}

### 例句
- {英文例句，造词的单词需要加粗}（句子的中文释义）{1...2}

### 拆分
- {拆分的单词}：拆分单词的中文释义 {1...n}

联想记忆：{拆分联想记忆}

### 同义
- {同义}({音标})：{同义的中文释义} {1...3}

### 反义
- {反义}({音标)：{反义的中文释义} {1...3}
`

export const VERIFY_SYSTEM_ROLE_TEMPLATE = "hello"
