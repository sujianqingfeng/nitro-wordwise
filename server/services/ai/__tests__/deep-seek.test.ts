import { test, describe, expect } from "bun:test"
import createDeepSeek from "../deep-seek"
import { SYSTEM_ROLE } from "./constants"

describe("deep-seek", () => {
	test(
		"generate",
		async () => {
			const apiKey = process.env.TEST_DEEP_SEEK_API_KEY

			if (!apiKey) {
				throw new Error("DEEP_SEEK_API_KEY is not set")
			}

			const text = await createDeepSeek({ apiKey }).generateText({
				system: SYSTEM_ROLE,
				prompt:
					"The code-first automation platform to seamlessly create and integrate custom JavaScript logic into any system, allowing you to easily extend your tools",
			})

			expect(text).toMatchSnapshot()
		},
		{
			timeout: 60 * 1000,
		},
	)
})
