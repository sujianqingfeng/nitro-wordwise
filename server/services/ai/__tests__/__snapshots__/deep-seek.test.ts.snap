// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`deep-seek generate 1`] = `
"### 分析
- 关键词组1:
  - 词组：code-first automation platform
  - 释义：一种以代码为核心的自动化平台，强调从代码开始构建自动化流程。
- 关键词组2:
  - 词组：seamlessly create and integrate
  - 释义：无缝地创建和集成，指过程流畅且不显眼。
- 关键词组3:
  - 词组：custom JavaScript logic
  - 释义：定制的JavaScript逻辑，指根据特定需求编写的JavaScript代码。
- 关键词组4:
  - 词组：extend your tools
  - 释义：扩展你的工具，指增加工具的功能或能力。

### 译文初稿
以代码优先的自动化平台，能够无缝地创建和集成定制的JavaScript逻辑到任何系统中，使你轻松扩展你的工具。

### 校对
- 校对意见1:
  - 原文：code-first automation platform
  - 译文：以代码优先的自动化平台
  - 问题：“以代码优先”表达略显生硬，不够自然。
  - 建议：改为“代码驱动的自动化平台”。
- 校对意见2:
  - 原文：seamlessly create and integrate
  - 译文：能够无缝地创建和集成
  - 问题：“能够”多余，直接用“无缝地”即可。
  - 建议：去掉“能够”。
- 校对意见3:
  - 原文：extend your tools
  - 译文：使你轻松扩展你的工具
  - 问题：“使你轻松”略显冗余，直接表达“扩展工具”即可。
  - 建议：简化为“扩展工具”。

### 译文终稿
代码驱动的自动化平台，无缝地创建和集成定制的JavaScript逻辑到任何系统中，轻松扩展工具。"
`;
